### PR TITLE
EOS-20652: HA: Refactor ha setup test command code

### PR DESCRIPTION
**Pull request by ajaysrivas**
## Problem Statement
<pre>
  <code>
  EOS-20652: HA: Refactor ha setup test command code
  Covered -
  1. Fixed all review comments from https://github.com/Seagate/cortx-ha/pull/382
  2. Wired the code to ha_setup test
  3. Component json files for every env - hw, vm, single-vm
  4. The components json files are not accurate and needs to be fixed as we progress.
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  1. Happy path on 3-node system is tested.
  </code>
</pre>

### CODE REVIEW

**Comment by ajaysrivas** [comment]
Ignore components files for review. The code in PR will look smaller.

File: ha/setup/ha_setup.py [hunk 1]
```diff
@@ -569,7 +570,28 @@ def process(self):
         """
         Process test command.
         """
-        pass # TBD: Write code here to check that all resources have been configured.
+        path_to_comp_config = const.SOURCE_CONFIG_PATH
+
+        hw_type = ConfigManager.get_hw_env()
```
Comment (ajay-paratmandali) [inline ha/setup/ha_setup.py:575]: Can we move get_env logic in parent Cmd class

```
        hw_type = ConfigManager.get_hw_env()
        if hw_type is not None:
            hw_type = hw_type.lower()
        else:
            Log.error(f"Error: Can not fetch h/w env from Config.")
            raise HaConfigException("h/w env not present in config.")

        nodes = self.get_nodelist(fetch_from=Cmd.HA_CONFSTORE)
        if len(nodes) == 1 and hw_type == const.INSTALLATION_TYPE.VM:
            hw_type = const.INSTALLATION_TYPE.SINGLE_VM

        Log.info(f"Nodes count = {len(nodes)}, Install type = {hw_type}")
```

**Comment by indrajitzagade** [comment]
@ajaysrivas Lets fix DCO and codacy issues.

**Comment by ajaysrivas** [comment]
@indrajitzagade Done.

**Review by indrajitzagade — APPROVED** [review]

**Pull request merged**